robolectric/src/test/java/org/robolectric/shadows/ShadowDialogTest.java

```diff
@@ -181,6 +181,19 @@ public void show_shouldWorkWithAPI19() {
     dialog.show();
   }
 
+  @Test
+  public void canSetAndGetOnCancelListener() {
+    Dialog dialog = new Dialog(RuntimeEnvironment.application);
+    DialogInterface.OnCancelListener onCancelListener = new DialogInterface.OnCancelListener() {
+      @Override
+      public void onCancel(DialogInterface dialog) {
+
+      }
+    };
+    dialog.setOnCancelListener(onCancelListener);
+    assertSame(onCancelListener, shadowOf(dialog).getOnCancelListener());
```

Comment:
Please use the AssertJ matchers (i.e `assertThat(foo).isSameAs(bar)`) instead of the Junit ones.